core/sql/optimizer/NATable.cpp

```diff
@@ -1472,58 +1478,161 @@ static ItemExpr * getRangePartitionBoundaryValuesFromEncodedKeys(
               // value. Note that the buffer has a prefix of some bytes with actual key
               // values, followed by bytes that are zeroed out. 
 
+              // the number of bytes actually provided in the key (not filled in)
+              int numBytesInProvidedVal = encodedKeyLen-valOffset;
+
+              if (nullHdrSize && numBytesInProvidedVal <= 0)
```

Comment:
The condition numBytesInProvidedVal <= 0 should always be true by definition of columnIsPartiallyProvided. Doesn't hurt anything though, just makes reading more complicated.